### ISSUE TITLE
How can we specify the scope?

### ISSUE THREAD

**Issue by dimacurotech** [jbogard/MediatR.Extensions.Microsoft.DependencyInjection#44]
Currently, AddMediatR uses a "Scoped" scope. How can one use Singleton or Transient scope?

**Comment by jbogard** [comment]
Yes, since this library is mostly intended to be used in ASP.NET Core, the appropriate lifecycle is "scoped". The only two options for ASP.NET Core for me are scoped or transient.

Singleton would carry the most unintended consequences. If you changed anything in this package to use singleton, then no other dependency under anything this resolves could be marked scoped. Singleton is really only appropriate for classes with no other dependencies (since you can't guarantee the lifecycle of dependencies).

Transient is a bit easier, as transients can depend on singleton/scoped items. However, it seems a bit weird to do so.

Where you might want to use singleton are in cases where you're using this package outside of ASP.NET Core. Is that what you're doing?
